BotModules/SteamBot/index.js

```diff
@@ -10,37 +10,68 @@ module.exports = function(bot) {
     helptext: 'Sends a game chosen randomly from a list of games shared between mentioned users',
     action: PickRandomSharedGame
+  },{
+    keyword: '/search',
+    helptext: 'Searches Games Collection for game that most closely matches message content. Limits results up to 3',
+    action: SearchGames
   }];
 
   bot.on('ready', () => {
     console.info(`Steambot logged in as ${bot.user.tag}!`);
   });
   
   bot.on("message", msg => {
-    let arrMentions = msg.mentions.users.map(user => user.id)
-    if(arrMentions.length > 1){
-      commands.forEach(command => {
-        if(msg.content.startsWith(command.keyword)){
-          command.action(msg, arrMentions);
-        }
-      });
-    }
+    commands.forEach(command => {
+      if(msg.content.startsWith(command.keyword)){
+        command.action(msg);
+      }
+    });
   });
   
-  function PickRandomSharedGame(msg, mentions){
-    MongoDB.GetCommonGames(mentions).then(games =>{
-      let rdmIdx = Math.floor(Math.random() * games.length)
-      msg.channel.send(`Lets play ${games[rdmIdx].name}!`)
-      MongoDB.Close()
-    })
-    .catch(err => console.log(err))
+  function PickRandomSharedGame(msg){
+    let mentions = msg.mentions.users.map(user => user.id)
+    if(arrMentions.length > 1){
+      MongoDB.GetCommonGames(mentions).then(games =>{
+        let rdmIdx = Math.floor(Math.random() * games.length)
+        msg.channel.send(`Lets play ${games[rdmIdx].name}!`)
+        MongoDB.Close()
+      })
+      .catch(err => console.log(err))
+    }
   };
 
-  function ShowAllSharedGames(msg, mentions){
-    MongoDB.GetCommonGames(mentions).then(games =>{
-      msg.channel.send(`${msg.mentions.users.map(user => user.username).join(", ")} share: \n${games.map(game => game.name).join("\n")}`)
-      MongoDB.Close()
+  function ShowAllSharedGames(msg){
+    let mentions = msg.mentions.users.map(user => user.id)
+    if(arrMentions.length > 1){
+      MongoDB.GetCommonGames(mentions).then(games =>{
+        msg.channel.send(`${msg.mentions.users.map(user => user.username).join(", ")} share: \n${games.map(game => game.name).join("\n")}`)
+        MongoDB.Close()
+      })
+      .catch(err => console.log(err))
+    }
+  };
+
+  function SearchGames(msg){
+    let searchText = msg.content.replace("/search", "").trim()
+    if(searchText === ""){
+      msg.channel.send("Invalid Search")
+      return
+    }
+    let pattern = new RegExp(escapeRegex(searchText), 'gi');
+    msg.channel.send("Searching games...")
+    MongoDB.SearchGamesFromText(pattern).then(games =>{
+      if(games.length == 0){
+        msg.channel.send("No Results Found")
+      }
+      else{
+        games.forEach(game => {
+          msg.channel.send(`${game.name}:\n\tIs owned by: ${game.members.map(member => member.alias).join(", ")}`)
+        })
+      }
     })
-    .catch(err => console.log(err))
+  }
+
+  function escapeRegex(text) {
+    return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
```

Comment:
I'll be honest.... I have no clue what this is... What MAGIC are you doing?

Reply:
Basically just takes user input and makes it into a regex friendly pattern that ignores special characters like spaces and other delimiters. Its a bit of overkill in this context, but it does cover all the bases. 
I may change this later on if we end up using only text indexes.